eth_utils/formatting.py

```diff
@@ -37,3 +41,11 @@ def is_prefixed(value, prefix):
     return value.startswith(
         force_bytes(prefix) if is_bytes(value) else force_text(prefix)
     )
+
+
+@curry
+def pad_bytes(fill_with, num_bytes, unpadded):
```

Comment:
Name ambiguity bothers me.  Does this pad to the left or the right?

Also, implementation-wise, I'd say to just create a fresh implementation that *only* handles bytes values because the other functions are making heavy use of `force_text/force_bytes`.